src/metabase/models/field_values.clj

```diff
@@ -234,14 +235,17 @@
   [field-or-field-id]
   (if-not (map? field-or-field-id)
     (let [field-id (u/the-id field-or-field-id)]
-      (recur (or (t2/select-one ['Field :base_type :visibility_type :has_field_values] :id field-id)
+      (recur (or (t2/select-one 'Field :id field-id)
                  (throw (ex-info (tru "Field {0} does not exist." field-id)
                                  {:field-id field-id, :status-code 404})))))
     (let [{base-type        :base_type
            visibility-type  :visibility_type
-           has-field-values :has_field_values} field-or-field-id]
+           has-field-values :has_field_values} field-or-field-id
+          table (t2/select-one 'Table :id (:table_id field-or-field-id))
+          engine (:engine (t2/select-one 'Database :id (:db_id table)))]
```

Comment:
We have `metabase.driver.util/database->driver` which is TTL memoized and can leverage MLv2 metadata providers if any are bound. 

At some point in the future we'll probably want to update sync code and stuff like this to use MetadataProviders too so this would future-proof us a little bit